src/app/(proper_react)/redesign/(authenticated)/user/dashboard/page.tsx

```diff
@@ -20,27 +17,23 @@ import { getLatestOnerepScan } from "../../../../../../db/tables/onerep_scans";
 export default async function DashboardPage() {
   const session = await getServerSession(authOptions);
   if (!session?.user?.subscriber?.id) {
-    throw new Error("No session");
+    return redirect("/");
   }
+
   const headersList = headers();
   const countryCode = getCountryCode(headersList);
 
   const result = await getOnerepProfileId(session.user.subscriber.id);
   const profileId = result[0]["onerep_profile_id"] as number;
-  const scanResult = await getLatestOnerepScan(profileId);
-  const scanResultItems = scanResult?.onerep_scan_results?.data ?? [];
-
   if (
-    !hasSetupOnerep(session?.user) &&
+    !profileId &&
```

Comment:
@flozia Isn't this the exact same logic as in `hasSetupOneReq`, but now no longer in a function name that describes why it does this check? (And also theoretically failing if the profile ID is 0, though I don't think that's possible.)